src/rampup.ts

```diff
@@ -0,0 +1,181 @@
+                const stats = await fs.promises.stat(filePath);
+    
+                if (stats.isFile() && (filePath.endsWith(extension) || filePath.endsWith('.js')) && !filePath.includes('/test/') && !filePath.includes('/node_modules/')) {
+                    const fileContent = await fs.promises.readFile(filePath, 'utf-8');
+                    const lines = fileContent.split('\n').length;
+                    totalLines += lines;
+                } else if (stats.isDirectory()) {
+                    // Recursively search for TypeScript files in subdirectories
+                    totalLines += await this.countLinesInDir(filePath, extension);
+                }
+            }
+    
+            return totalLines;
+        } catch (error) {
+            throw error;
+        }
+    }
+    
+    async findReadmeFile(directory: string): Promise<string | null> {
+        try {
+            const files = await fs.promises.readdir(directory);
+    
+            for (const file of files) {
+                const filePath = path.join(directory, file);
+    
+                if (file.toLowerCase() === 'readme.md' || file.toLowerCase() === 'readme.txt') {
+                    return filePath;
+                } else if ((await fs.promises.stat(filePath)).isDirectory()) {
+                    const readmePath = await this.findReadmeFile(filePath);
+                    if (readmePath) {
+                        return readmePath;
+                    }
+                }
+            }
+    
+            return null;        
+        } catch (error) {
+            throw error;
+        }
+    }
+    
+    async countLinesInReadme (path : string): Promise <number> {
+        let lines = 0;
+        
+        try {
+            const readmeContent = await fs.promises.readFile(path, 'utf-8');
+            lines = readmeContent.split('\n').length;
+        } catch(error) {
+            throw error;
+        }
+    
+        return lines;
+    }
+    
+    async rampup():Promise <number> {
+        const repoDirectory = '/Users/junpeiota/nest'; // Replace with the path to your cloned repository directory
+        const extension = '.js'; // Change the file extension as needed
+        const scale = 10; //How many lines of code in files to one line in the Readme
+        let linesInFiles = 0; //LinesInFiles
+        let linesInReadme = 0; //Lines in README
+        let metric: number;
+    
+        try {
+            linesInFiles = await this.countLinesInDir(repoDirectory, extension);
+            const readmePath = await this.findReadmeFile(repoDirectory);
+    
+            if (readmePath) {
+                const linesInReadme = await this.countLinesInReadme(readmePath);
+                console.log(`Total lines in TypeScript files: ${linesInFiles}`);
+                console.log(`Total lines in README: ${linesInReadme}`);
+    
+                metric = Math.max(linesInFiles / (linesInReadme * scale ), 1);
```

Comment:
Will this have a value greater than 1? 
If the lines in readme * scale is still less than linesInFiles，metric could have a greater than 1.

Should we do Math.min here instead?

Reply:
Yeah you're right. So the change would be to Math.min(linesInFiles / (linesInReadme * scale ), 1). linesInFIles should never be 0 and linesInREadme should also never be 0 since we return if we do not find it. As a result those variables should always be positive and we should never get a value smaller than 0. In case the linesInFiles / (linesInReadme * scale is greater than 1 then we should take the min of 1 to ensure we stay within the bounds of [0,1]